mmv1/products/netapp/volume.yaml

```diff
@@ -0,0 +1,412 @@
+      - :SERVICE_MANAGED
+      - :CLOUD_KMS
+    output: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'hasReplication'
+    description: |
+      Indicates whether the volume is part of a volume replication relationship.
+    output: true
+  - !ruby/object:Api::Type::Array
+    name: 'restrictedActions'
+    description: |
+      List of actions that are restricted on this volume.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'restrictedActions'
+      description: |
+        Settings for volumes with SMB access.
+      values:
+        - :RESTRICTED_ACTION_UNSPECIFIED
+        - :DELETE
+  - !ruby/object:Api::Type::Array
+    name: 'mountOptions'
+    description: Reports mount instructions for this volume.
+    output: true
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'export'
+          description: Export path of the volume.
+          output: true
+        - !ruby/object:Api::Type::String
+          name: 'exportFull'
+          output: true
+          description: |
+            Full export path of the volume.
+
+            Format for NFS volumes: `<export_ip>:/<shareName>`
+
+            Format for SMB volumes: `\\\\netbios_prefix-four_random_hex_letters.domain_name\\shareName`
+        - !ruby/object:Api::Type::String
+          name: 'instructions'
+          description: Human-readable mount instructions.
+          output: true
+        - !ruby/object:Api::Type::Enum
+          name: 'protocol'
+          description: |
+              Protocol to mount with.
+          values:
+            - :PROTOCOLS_UNSPECIFIED
+            - :NFSV3
+            - :NFSV4
+            - :SMB
+          output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: snapshotPolicy
+    description: |-
+      Snapshot policy defines the schedule for automatic snapshot creation.
+      To disable automatic snapshot creation you have to remove the whole snapshot_policy block.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |-
+          Enables automated snapshot creation according to defined schedule. Default is false.
+          To disable automatic snapshot creation you have to remove the whole snapshot_policy block.
+        default_value: false
+        send_empty_value: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'hourlySchedule'
+        description: |-
+          Hourly schedule policy.
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'snapshotsToKeep'
+            description: |-
+              The maximum number of snapshots to keep for the hourly schedule.
+            required: true
+          - !ruby/object:Api::Type::Integer
+            name: 'minute'
+            description: |-
+              Set the minute of the hour to create the snapshot (0-59), defaults to the top of the hour (0).
```

Comment:
Do we want to add client side default value for these subfields of `*Schedule`? Will the API return the default values if we don't explicitly specify the fields

Reply:
Good point. Adding default values for all the fields which says "defaults to xxx".

Reply:
Also removed the field from one of the tests to see if the defaults work.